solr/core/src/test/org/apache/solr/client/solrj/cloud/autoscaling/MoveReplicaSuggesterTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.client.solrj.cloud.autoscaling;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.solr.common.cloud.Replica;
+import org.apache.solr.common.cloud.ZkStateReader;
+import org.apache.solr.common.util.Pair;
+import org.junit.Test;
+
+import static org.apache.solr.client.solrj.cloud.autoscaling.MoveReplicaSuggester.leaderLast;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class MoveReplicaSuggesterTest {
+  private String CORE = "collection_shard1_replica_n1";
+  private String COLLECTION = "collection";
+  private String SHARD = "shard1";
+  private Map<String, Object> IS_LEADER = new HashMap<String, Object>() {{ put(ZkStateReader.LEADER_PROP, true);}};
```

Comment:
NABD, but you can do `Collections.singletonMap(ZkStateReader.LEADER_PROP, true);`

Reply:
Thanks for the suggestion, I've done this now. Being curious, what advantage does this give? Is it for memory management or?

Reply:
In this case brevity (as it's a test).  Plus the way you were doing it is actually generating a new anonymous class.  It's annoying that Java doesn't have Python style native syntax for creating maps.